aten/src/ATen/native/TensorShape.cpp

```diff
@@ -121,19 +124,21 @@ Tensor & _cat_out_cpu(Tensor& result, TensorList tensors, int64_t dim) {
   }
   at::assert_no_internal_overlap(result);
 
-  auto should_skip = [](const Tensor& t) { return t.numel() == 0 && t.dim() == 1; };
-  for (auto const &tensor : tensors) {
-    if (should_skip(tensor)) {
-      continue;
+  const Tensor* pnotSkippedTensor = [](const TensorList& tensors) -> const Tensor* {
```

Comment:
@swolchok It would be nice here to have a comment that explains why it is ok to take a plain pointer here and why we are sure all the Tensors are kept alive until the end of this function.

Reply:
Again, this is my change, but I agree with you, a comment will be nice.

Reply:
1) there's no reason to pass TensorList by reference; it's a reference type already
2) If a reference is OK, a pointer is OK

Reply:
> If a reference is OK, a pointer is OK

I am sure it is ok, I am just saying that we should usually have a good reason before using plain pointers. So having a comment here will help future readers understand easily why we do it here.